rewards chart aggregation

enzo-bitfly opened the pull request.

enzo-bitfly committed:
refactor: expose `has hability chart history` flag from `user` store instead of `dashboard overview`store

`chart history seconds` is part of the `user` and `dashboard overview` data, but `rewards_history_seconds` is only part of `user` data

See: BEDS-875

marcel-bitfly commented on frontend/assets/css/prime.scss:
```diff
@@ -626,6 +626,122 @@ div.p-accordion {
   }
 }
 
+div.p-datepicker-input {
```
**suggestion:** ideally all styles related to the datepicker should be located in `BcDatePicker.vue`

So we could one day remove `prime.scss` (since these styles get loaded on every page)

enzo-bitfly replied:
yeah did this before wrapping the date picker and forgot to move it there. Done

marcel-bitfly commented on frontend/components/bc/BcDatepicker.vue:
```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+const {
+  dateFormat = 'dd/mm/yy',
```
**issue:** `date format` could be ambiguous for users (e.g. USA: mm/dd/yyy)

this is how we are formatting the date currently:
<img width="1367" height="760" alt="Bildschirmfoto 2025-09-01 um 12 52 53" src="https://github.com/user-attachments/assets/50890e69-be71-4827-8dfa-3cac54bceadc" />

enzo-bitfly replied:
thanks, done

marcel-bitfly commented on frontend/components/bc/BcDatepicker.vue:
```diff
@@ -0,0 +1,42 @@
+}>()
+
+const modelValue = defineModel<Date>({
+
```
**nitpick:** 👀

enzo-bitfly replied:
whoops

marcel-bitfly commented on frontend/components/dashboard/chart/DashboardChartRewards.vue:
```diff
@@ -446,6 +449,8 @@ const option = computed<EChartsOption>(() => {
   }
 })
+
+onUpdated(() => hideTooltip())
```
**suggestion:** maybe try to use an event that is exposed by the chart itself instead

enzo-bitfly replied:
not needed anymore after making the tooltip appear on hover

enzo-bitfly reviewed (commented):
great review, thanks 💪

enzo-bitfly committed:
feat: add `rewards chart` filters

- prevent showing `aggregation` filter in `summary` chart to match `rewards chart`
- fix aggregation filter for logged-out usrers in `summary chart`

See: BEDS-875
